lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerendpoint.py

```diff
@@ -0,0 +1,350 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Hai Cao, <t-haicao@microsoft.com>, Yunge Zhu <yungez@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_trafficmanagerendpoint
+version_added: "2.7"
+short_description: Manage Azure Traffic Manager endpoint.
+description:
+    - Create, update and delete Azure Traffic Manager endpoint.
+
+options:
+    resource_group:
+        description:
+            - Name of a resource group where the Traffic Manager endpoint exists or will be created.
+        type: str
+        required: true
+    name:
+        description:
+            - The name of the endpoint.
+        type: str
+        required: true
+    profile_name:
+        description: Name of Traffic Manager profile where this endpoints attaches to.
+        type: str
+        required: true
+    type:
+        description:
+            - The type of the endpoint.
+        required: true
+        choices:
+            - azureEndpoints
+            - externalEndpoints
+            - nestedEndpoints
+    target_resource_id:
+        description:
+            - The Azure Resource URI of the of the endpoint.
+            - Not applicable to endpoints of type 'ExternalEndpoints'.
+        type: str
+    target:
+        description:
+            - The fully-qualified DNS name of the endpoint.
+        type: str
+    enable:
+        description:
+            - The status of the endpoint.
+        type: bool
+        default: true
+    weight:
+        description:
+            - The weight of this endpoint when using the 'Weighted' traffic routing method.
```

Comment:
refer to the actual options in the traffic manager profile module and not just a copy/paste of the MS docs. In this case it should be something like

`The weight of the endpoint when the traffic profile has a routing_method of C(weighted)`

This has the benefit of showing both the trafficmanagerprofile method name (routing_method) and the value of it in Ansible.

Reply:
fixed